conans/client/cmd/export.py

```diff
@@ -296,31 +308,27 @@ def _replace_scm_data_in_conanfile(conanfile_path, scm_data):
                     line = lines[stmt.lineno - 1]
                     tab_size = len(line) - len(line.lstrip())
                     if isinstance(stmt.targets[0], ast.Name) and stmt.targets[0].id == "scm":
-                        try:
-                            if i + 1 == len(statements):  # Last statement in my ClassDef
-                                if i_body + 1 == len(tree.body):  # Last statement over all
-                                    next_line = len(lines)
-                                else:
-                                    next_line = tree.body[i_body+1].lineno - 1
-                            else:
-                                next_line = statements[i+1].lineno - 1
-                                next_line_content = lines[next_line].strip()
-                                if (next_line_content.endswith('"""') or
-                                        next_line_content.endswith("'''")):
-                                    next_line += 1
-                        except IndexError:
-                            next_line = stmt.lineno
-                        replace = [line for line in lines[(stmt.lineno-1):next_line]]
-                        to_replace.append("".join(replace).lstrip())
-                        comments = [line.strip('\n') for line in replace
-                                    if line.strip().startswith("#") or not line.strip()]
+                        to_replace = [ast.get_source_segment(content, stmt), ]
```

Comment:
This is py38, this deserves its own PR just refactoring it. For previous versions of Python, we can copy/paste the implementation, it really simplifies the source code and fixes all the `FIXME` that were pending.